dbt_style_guide.md

```diff
@@ -1,237 +1,415 @@
+Example:
+  ```sql
+  where 
+    user_id is not null
+    and status = 'pending'
+    and location = 'hq'
+  ```
+
+- Joins should list the "left" table first (i.e., the table you're joining data to):
+  ```sql
+  select
     trips.*,
     drivers.rating as driver_rating,
     riders.rating as rider_rating
-
-from trips
-left join users as drivers
+  from trips
+  left join users as drivers
     on trips.driver_id = drivers.user_id
-left join users as riders
+  left join users as riders
     on trips.rider_id = riders.user_id
 
-```
+  ```
+
+- ### Example SQL
+  ```sql
+  with
+
+  my_data as (
+      select * from {{ ref('my_data') }}
+      where not is_deleted
+  ),
+
+  some_cte as (
+      select * from {{ ref('some_cte') }}
+  ),
+
+  some_cte_agg as (
+      select
+          id,
+          sum(field_4) as total_field_4,
+          max(field_5) as max_field_5
+      from some_cte
+      group by 1
+  ),
+
+  final as (
+      select [distinct]
+          my_data.field_1,
+          my_data.field_2,
+          my_data.field_3,
+
+          -- use line breaks to visually separate calculations into blocks
+          case
+              when my_data.cancellation_date is null
+                and my_data.expiration_date is not null
+                then expiration_data
+              when my_data.cancellation_date is null
+                then my_data.start_date + 7
+              else my_data.cancellation_date
+          end as cancellation_date,
+
+          some_cte_agg.total_field_4,
+          some_cte_agg.max_field_5
+      from my_data
+      left join some_cte_agg  
+          on my_data.id = some_cte_agg.id
+      where 
+        my_data.field_1 = 'abc'
+        and (
+            my_data.field_2 = 'def'
+            or my_data.field_2 = 'ghi'
+        )
+      having count(*) > 1
+  )
+
+  select * from final
+  ```
 
 ## YAML style guide
 
-* Indents should be two spaces
-* List items should be indented
-* Use a new line to separate list items that are dictionaries where appropriate
-* Lines of YAML should be no longer than 80 characters.
-
-### Example YAML
-```yaml
-version: 2
-
-models:
-  - name: events
-    columns:
-      - name: event_id
-        description: This is a unique identifier for the event
-        tests:
-          - unique
-          - not_null
-
-      - name: event_time
-        description: "When the event occurred in UTC (eg. 2018-01-01 12:00:00)"
-        tests:
-          - not_null
-
-      - name: user_id
-        description: The ID of the user who recorded the event
-        tests:
-          - not_null
-          - relationships:
-              to: ref('users')
-              field: id
-```
+- Every subdirectory contains their own `.yml` file(s) which contain configurations for the models within the subdirectory.
+
+- A separate `.yml` file is created per top-level configuration (`sources`, `models`) that applies to the models within the subdirectory. 

```

Comment:
The duplicate entry was removed, and I think that we should refrain from explaining _another_ practice that isn't the default (so that would change to "We recommend creating ...". The example yml location can already be found in the Best Practices article we linked above, so I personally feel that we should omit a file structure example. 

Gonna leave this one open to see what others think! The guide is getting pretty long so another thing I have on the brain is trying to avoid being too verbose, but since I'm in this headspace of many changes I'd really prefer if everyone else came at this from a different angle!

Reply:
My reasoning for the example is because I had to read the sentence a few times to figure out what it meant. If someone is less familiar than us, they might need the example. 

I'd say this is a great time to be verbose!

Reply:
Vote to keep in the examples! I always appreciate them and will just scroll past if I don't need the extra help.

Reply:
I'm good either way on an example or not, but if we keep the example I think it needs tweaked a bit. Right now it's showing the `jaffle_shop` folder as an empty folder, with the files at the same level as the folder. The files from `_jaffle_shop__models.yml` to `stg_jaffle_shop__payments.sql` should be one level under `jaffle_shop`.

Reply:
Added yml example and fixed the filetree [here](https://github.com/dbt-labs/corp/pull/63/commits/26274f7b5939791e82afcbd4af10498086a03ed8)!